### DISCUSSION TITLE
如何作为@ice/stark的子应用

### DISCUSSION THREAD

**Discussion by xmsz-stu** [web-infra-dev/modern.js#6053] [Q&A]
想把旧项目切换到modern.js
主应用目前用的是@ice/stark
子应用用modernjs

现在想要把子应用接入到主应用中，不知道这么做

**Comment by zllkjc** [comment]
可以提供之前的接入方式，或是这个主应用需要 modern.js 支持什么样的能力

**Reply by xmsz-stu** [reply]
之前通过https://v2.ice.work/docs/guide/advanced/icestark，这样方式接入，我现在的配置

```
{
       html: {
      crossorigin: 'anonymous',
      disableHtmlFolder: true,
    },
    plugins: [
      appTools({
        bundler: 'experimental-rspack',
      }),
      garfishPlugin(),
    ],
   deploy: {
      microFrontend: true,
    },
  source: {
      mainEntryName: 'index',
    },
}
```
然后遇到的问题
- 因为ice也会拦截history，导致garfish的normalAgent失效，我现在延迟执行，但是很麻烦
- 项目用unocss，但是打包的文件会在static/index.css，导致子应用热更新后样式没变化，我现在给每个page添加unocss.css才行，但是不合理
